File: primos_gemeos_numeros.py
def eh_primo(n):
    """Verifica se um número é primo."""
    if n < 2:
        return False
    for i in range(2, int(n**0.5) + 1):
        if n % i == 0:
            return False
    return True

def encontrar_primos_gemeos(limite):
    """Encontra todos os pares de primos gêmeos até o limite estipulado."""
    primos_gemeos = []

    # Começamos do 3, pois o único primo par é 2, e ele não forma gêmeo com ninguém acima de 3.
    for n in range(3, limite - 1, 2):
        if eh_primo(n) and eh_primo(n + 2):
            primos_gemeos.append((n, n + 2))
            
    return primos_gemeos

# --- Execução do Programa ---
if __name__ == "__main__":
    limite_usuario = int(input("Digite o limite máximo para buscar primos gêmeos: "))
    
    resultado = encontrar_primos_gemeos(limite_usuario)
    
    print(f"\nEncontrados {len(resultado)} pares de primos gêmeos até {limite_usuario}:")
    for par in resultado:
        print(f"{par[0]} e {par[1]}")

File: primos_gemeos_numeros.c
#include <stdio.h>
#include <stdbool.h>
#include <math.h>

// Função para verificar se um número é primo
bool eh_primo(int n) {
    if (n <= 1) return false;
    if (n == 2) return true;
    if (n % 2 == 0) return false; // Elimina os pares de cara

    // Testa os divisores ímpares até a raiz quadrada de n
    int limite = (int)sqrt(n);
    for (int i = 3; i <= limite; i += 2) {
        if (n % i == 0) {
            return false;
        }
    }
    return true;
}

int main() {
    int limite;
    int contador = 0;

    printf("Digite o limite maximo para buscar primos gemeos: ");
    if (scanf("%d", &limite) != 1) {
        printf("Entrada invalida.\n");
        return 1;
    }

    printf("\nPares de primos gemeos ate %d:\n", limite);
    printf("-----------------------------------\n");

    // Começa do 3, já que (3, 5) é o primeiro par de primos gêmeos
    for (int i = 3; i <= limite - 2; i += 2) {
        // Se 'i' e 'i + 2' forem primos, temos um par gêmeo
        if (eh_primo(i) && eh_primo(i + 2)) {
            printf("(%d, %d)\n", i, i + 2);
            contador++;
        }
    }

    printf("-----------------------------------\n");
    printf("Total de pares encontrados: %d\n", contador);

    return 0;
}